src/components/calltree/ProfileCallTreeContextMenu.js

```diff
@@ -167,14 +163,58 @@ class ProfileCallTreeContextMenu extends PureComponent {
       case 'merge-function':
         addTransformToStack(threadIndex, {
           type: 'merge-function',
-          funcIndex: selectedCallNodePath[selectedCallNodePath.length - 1],
+          funcIndex: selectedFunc,
         });
         break;
+      case 'collapse-resource': {
+        const { funcTable } = thread;
+        const resourceIndex = funcTable.resource[selectedFunc];
+        // A new collapsed func will be inserted into the table at the end. Deduce
+        // the index here.
+        const collapsedFuncIndex = funcTable.length;
+        addTransformToStack(threadIndex, {
+          type: 'collapse-resource',
+          resourceIndex,
+          collapsedFuncIndex,
+          implementation,
+        });
+        break;
+      }
       default:
         throw new Error('Type not found.');
     }
   }
 
+  getNameForResouce(): string | null {
```

Comment:
Call this `getNameForSelectedResource()`? I was looking for a `resource` argument here for a second.
Alternatively, you could pass the funcIndex which you already seem to have in the caller.